src/ucs/arch/aarch64/cpu.h

```diff
@@ -29,10 +29,10 @@ BEGIN_C_DECLS
 #define ucs_memory_bus_fence()        asm volatile ("dsb sy" ::: "memory");
 #define ucs_memory_bus_store_fence()  asm volatile ("dsb st" ::: "memory");
 #define ucs_memory_bus_load_fence()   asm volatile ("dsb ld" ::: "memory");
-
 #define ucs_memory_cpu_fence()        asm volatile ("dmb ish" ::: "memory");
 #define ucs_memory_cpu_store_fence()  asm volatile ("dmb ishst" ::: "memory");
 #define ucs_memory_cpu_load_fence()   asm volatile ("dmb ishld" ::: "memory");
+#define ucs_memory_cpu_wc_fence()     asm volatile ("dmb st" ::: "memory");
```

Comment:
I need we need more than simple fence.
We also need something like wc_64b_retire and wc_8b_retire.
On x86 64b is nop and 8b will be mapped to fence.
On aarch64 it will be based on uarch that is set during configure.

Reply:
currently there is no 64b retire; we assume (maybe wrongfully, but it works) that 64b are retired automatically by u-arch. it was not introduced by this PR

Reply:
You actually will never know if it works or not. uarch will retire it on timemout.
Regardless, since you already cleaning up the code I suggest to introduce the macro.
Feel free to map to "nop" on x86.

Reply:
this PR is for fixing specific thread safety issue, and i don't see why it should depend on doing additional cleanup in the code